Source/content/en/docs/tutorials/event-horizon.md

```diff
@@ -0,0 +1,496 @@
+
+    event-horizon-tutorial
+        consumer
+            ...
+        producer
+            ...
+        environment
+            docker-compose.yml
+            ...
+
+Go into both the *consumer* and the *producer* folders and initiate the project as we've gone through in our [getting started]({{< ref "docs/tutorials/getting_started" >}}) guide. I.e copy over all the code from the getting started tutorial to the *consumer* and *producer* folders.
+
+We'll come back to the docker-compose later in this tutorial.
+
+## Producer
+### Create a `Public Filter`
+A `public filter` filters all events that are committed as public events into a [`public stream`]({{< ref "docs/concepts/streams" >}}), which is special stream that another microservice can subscribe to. A public filter definition in the SDK's perspective is simply a method that returns a partitioned filter result which simply is an object with two properties: a boolean that says whether the event should be included in the stream and a partition id which is the partition that the event should belong to in the stream. Only the events that are committed as a public events gets filtered through the public filters.  
+
+{{< tabs name="public-filter-tab" >}}
+{{% tab name="C#" %}}
+```csharp
+// Program.cs
+using System;
+using System.Threading.Tasks;
+using Dolittle.SDK;
+using Dolittle.SDK.Tenancy;
+using Dolittle.SDK.Events;
+using Dolittle.SDK.Events.Filters;
+
+namespace Kitchen
+{
+    class Program
+    {
+        public static void Main()
+        {
+            var client = Client
+                .ForMicroservice("f39b1f61-d360-4675-b859-53c05c87c0e6")
+                .WithEventTypes(eventTypes =>
+                    eventTypes.Register<DishPrepared>())
+                .WithEventHandlers(builder =>
+                    builder.RegisterEventHandler<DishHandler>())
+                .WithFilters(filtersBuilder =>
+                    filtersBuilder
+                        .CreatePublicFilter("2c087657-b318-40b1-ae92-a400de44e507", filterBuilder =>
+                            filterBuilder.Handle((@event, eventContext) =>
+                            {
+                                Console.WriteLine($"Filtering event {@event} to public stream");
+                                return Task.FromResult(new PartitionedFilterResult(true, PartitionId.Unspecified));
+                            })))
+                .Build();
+            // Rest of your code here...
+        }
+    }
+}
+```
+{{% /tab %}}
+
+{{% tab name="TypeScript" %}}
+```typescript
+// index.ts
+import { Client } from '@dolittle/sdk';
+import { TenantId } from '@dolittle/sdk.execution';
+import { DishPrepared } from './DishPrepared';
+import { DishHandler } from './DishHandler';
```

Comment:
Should also import the `PartitionId`, ties into that we should update the samples in the SDK's to check for code errors

Reply:
Should be fixed by https://github.com/dolittle/Documentation/pull/83/commits/4ffdbfcd742bd9e386d6f7d922c76b5a63ee7406